pkg/blobstore/local/file_backed_location_record_array.go

```diff
@@ -0,0 +1,88 @@
+package local
+
+import (
+	"encoding/binary"
+	"sync"
+
+	"github.com/buildbarn/bb-storage/pkg/filesystem"
+)
+
+func serializeLocationRecord(locationRecord LocationRecord) []byte {
+	serialised := make([]byte, 60)
+	for i := 0; i < 32; i++ {
+		serialised[i] = locationRecord.Key.Digest[i]
+	}
+	binary.LittleEndian.PutUint32(serialised[32:36], locationRecord.Key.Attempt)
+	binary.LittleEndian.PutUint64(serialised[36:44], uint64(locationRecord.Location.BlockID))
+	binary.LittleEndian.PutUint64(serialised[44:52], uint64(locationRecord.Location.OffsetBytes))
+	binary.LittleEndian.PutUint64(serialised[52:60], uint64(locationRecord.Location.SizeBytes))
+
+	return serialised
+}
+
+func deserializeLocationRecord(record []byte) LocationRecord {
+	var keyDigest [32]byte
+	copy(keyDigest[0:32], record[0:32])
```

Comment:
Keep in mind that `copy()` copies `min(len(a), len(b))` bytes. It's therefore perfectly fine to just call:
```go
copy(keyDigest, record)
```
That said, I guess it would make a lot of sense to write all of this in a way that's a bit more dense.
```go
l := LocationRecord{
        Key: LocationRecordKey{
                Attempt: binary.LittleEndian.Uint32(record[32:]),
        },
        BlockID: int(binary.LittleEndian.Uint64(record[36:]),
        OffsetBytes: int64(binary.LittleEndian.Uint64(record[44:]))
        SizeBytes: int64(binary.LittleEndian.Uint64(record[52:]))
}
copy(l.Key.Digest, record)
```

You're only using these serializing/deserializing functions in one place. I guess readability wouldn't suffer if you simply inlined this into `Get()` and `Put()`.

Reply:
I've inlined the serializing/deserializing functions, which I think has probably improved readability actually. For the `copy` I still had to slice the arrays, but I removed the limits.